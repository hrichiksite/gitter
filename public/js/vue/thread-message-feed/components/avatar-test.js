jest.mock('gitter-web-avatars');

const mount = require('../../__test__/vuex-mount');
const avatars = require('gitter-web-avatars');
const { default: Avatar } = require('./avatar.vue');

describe('thread-message-feed avatar', () => {
  it('default size matches snapshot', () => {
    const { wrapper } = mount(Avatar, {
      user: {
        displayName: 'Test User',
        gravatarImageUrl:
          'https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=60&d=identicon'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('m size matches snapshot', () => {
    const { wrapper } = mount(Avatar, {
      size: 'm',
      user: {
        displayName: 'Test User',
        gravatarImageUrl:
          'https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=60&d=identicon'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use fallback image if there is a loading error', () => {
    avatars.getDefault.mockImplementation(() => 'https://default.png');
    const { wrapper } = mount(Avatar, {
      user: {
        displayName: 'Test User',
        gravatarImageUrl:
          'https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=60&d=identicon'
      }
    });
    wrapper.find('img').trigger('error');
    expect(wrapper.element).toMatchSnapshot();
  });
});
