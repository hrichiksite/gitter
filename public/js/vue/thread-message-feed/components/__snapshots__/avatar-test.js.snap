// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed avatar default size matches snapshot 1`] = `
<div
  class="trpDisplayPicture avatar-s"
>
  <img
    alt="Test User"
    class="avatar__image"
    src="https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=30&d=identicon"
    srcset="https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=60&d=identicon 2x"
  />
</div>
`;

exports[`thread-message-feed avatar m size matches snapshot 1`] = `
<div
  class="trpDisplayPicture avatar-m"
>
  <img
    alt="Test User"
    class="avatar__image"
    src="https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=48&d=identicon"
    srcset="https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=96&d=identicon 2x"
  />
</div>
`;

exports[`thread-message-feed avatar should use fallback image if there is a loading error 1`] = `
<div
  class="trpDisplayPicture avatar-s"
>
  <img
    alt="Test User"
    class="avatar__image"
    src="https://default.png"
  />
</div>
`;
